### PR TITLE
Add error pages

### LINKED ISSUE #322: 404 Error Page
**Is your feature request related to a problem? Please describe.**
When visiting a link nothing really happens right now. This can be a bit misleading.

**Describe the solution you'd like**
Show a nice (maybe even funny) 404 page.

### PR DESCRIPTION
**Pull request by joschahenningsen**
Resolves #322 

This Pr adds some error pages:
- 404 catch all
- 404 course not found
- 404 stream not found
- 403 stream/course access not allowed
- 403 admin access not allowed

https://user-images.githubusercontent.com/44805696/158150529-b6deb27e-9ca4-43da-a641-67c960fb060c.mp4

### CODE REVIEW

File: web/template/error.gohtml [hunk 1]
```diff
@@ -1,17 +1,43 @@
...
+            content: ' ';
+            display: block;
+            position: absolute;
+            left: 0;
+            top: 0;
+            width: 100%;
+            height: 100%;
+            z-index: 1;
+            opacity: 0.4;
+            background-image: url('https://media2.giphy.com/media/3o7aTskHEUdgCQAXde/giphy.gif');
```
Comment (joschahenningsen) [inline web/template/error.gohtml:24]: Let's see if this URL ever changes

File: dao/courses.go [hunk 2]
```diff
@@ -225,12 +225,12 @@ func GetAvailableSemesters(c context.Context) []Semester {
 }
 
 func GetCourseByShortLink(link string) (model.Course, error) {
```
Comment (alexanderstephan) [inline dao/courses.go:228]: This should probably have a comment.

File: web/template/error.gohtml (hunk 1 above)
Comment (alexanderstephan) [reply web/template/error.gohtml:24]: If it breaks, it's probably time for a new image. :joy:

File: tools/middlewares.go [hunk 3]
```diff
@@ -39,6 +47,34 @@ func InitContext(c *gin.Context) {
...
+
+const (
+	PageNotFoundErrMsg     = "This page does not exist."
+	CourseNotFoundErrMsg   = "We couldn't find the course you were looking for."
+	StreamNotFoundErrMsg   = "We couldn't find the stream you were looking for."
+	ForbiddenGenericErrMsg = "You don't have permissions to access this resource. " +
+		"Please reach out if this seems wrong :)"
+	ForbiddenStreamAccess = "You don't have permissions to access this stream. " +
+		"Please make sure to use the correct login."
+	ForbiddenCourseAccess = "You don't have permissions to access this stream. " +
```
Comment (alexanderstephan) [inline tools/middlewares.go]:
```suggestion
	ForbiddenCourseAccess = "You don't have permissions to access this course. " +
```

**Review by alexanderstephan — CHANGES_REQUESTED** [review]
Exactly what I imagined, really nice! :partying_face: 

I only noticed that some comments for exported methods might be needed, and a copy-paste problem. Otherwise the code changes look really clean. :+1:

**Review by alexanderstephan — APPROVED** [review]
Looks good to me! :100:

**Pull request merged**